src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs

```diff
@@ -136,7 +136,12 @@ private static PrimarySignature CreatePrimarySignature(CmsSigner cmsSigner, Sign
 
             try
             {
+#if IS_DESKTOP
                 cms.ComputeSignature(cmsSigner);
+#else
+                // In .NET Framework, this parameter is not used and a PIN prompt is always shown. In .NET Core, the silent flag needs to be set to false to show a PIN prompt.
+                cms.ComputeSignature(cmsSigner, silent: false);
+#endif
             }
             catch (CryptographicException ex) when (ex.HResult == INVALID_PROVIDER_TYPE_HRESULT)
```

Comment:
I think this is out of scope for this PR. may be we need a tracking issue to avoid depending on `HRResult`.

Reply:
Thanks! https://github.com/NuGet/Home/issues/9628 raise.